packages/qwik/src/optimizer/src/plugins/rollup.ts

```diff
@@ -168,59 +168,78 @@ export function normalizeRollupOutputOptions(
   path: Path,
   opts: NormalizedQwikPluginOptions,
   rollupOutputOpts: Rollup.OutputOptions | Rollup.OutputOptions[] | undefined
-) {
-  const outputOpts: Rollup.OutputOptions = {};
-  if (rollupOutputOpts && !Array.isArray(rollupOutputOpts)) {
-    Object.assign(outputOpts, rollupOutputOpts);
+): Rollup.OutputOptions[] {
+  const outputOpts: Rollup.OutputOptions[] = Array.isArray(rollupOutputOpts)
+    ? // fill the `outputOpts` array with all existing option entries
+      [...rollupOutputOpts]
+    : // use the existing rollupOutputOpts object or create a new one
+      [rollupOutputOpts || {}];
+
+  // make sure at least one output is present in every case
+  if (!outputOpts.length) {
+    outputOpts.push({});
   }
-  if (!outputOpts.assetFileNames) {
-    outputOpts.assetFileNames = 'build/q-[hash].[ext]';
+
+  return outputOpts.map((outputOptsObj) =>
+    normalizeRollupOutputOptionsObject(path, opts, outputOptsObj)
+  );
+}
+
+export function normalizeRollupOutputOptionsObject(
+  path: Path,
+  opts: NormalizedQwikPluginOptions,
+  rollupOutputOptsObj: Rollup.OutputOptions | undefined
+): Rollup.OutputOptions {
+  const outputOptsObj: Rollup.OutputOptions = { ...rollupOutputOptsObj };
+
+  if (!outputOptsObj.assetFileNames) {
+    outputOptsObj.assetFileNames = 'build/q-[hash].[ext]';
```

Comment:
Generally yes, but since I used the logic of this function pretty much 1:1 (I just removed the `Object.assign()` call in the beginning and renamed the function from `normalizeRollupOutputOptions` to `normalizeRollupOutputOptionsObject`), I would not like to make this change